Blog.API/Blog.Infrastructure/Services/CategoryService.cs

```diff
@@ -1,36 +1,125 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Threading.Tasks;
+﻿using Blog.Core.Contracts.Controllers.Categories;
 using Blog.Core.Entities;
+using Blog.Core.ResponseDtos;
+using Blog.Infrastructure.Abstract.Interfaces;
 using Blog.Infrastructure.Services.Interfaces;
+using FluentValidation;
+using MapsterMapper;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
 
 namespace Blog.Infrastructure.Services;
 
-//TODO:Implement methods
 public sealed class CategoryService : ICategoryService
 {
-    public IEnumerable<Category> GetCategories()
+    private readonly IUnitOfWork<Category> _unitOfWork;
+    private readonly IValidator<CreateCategoryRequest> _createCategoryValidator;
+    private readonly IValidator<UpdateCategoryRequest> _updateCategoryValidator;
```

Comment:
I'm not sure if we need to validate requests on service level. I think we should do it on request level